activemodel/test/cases/secure_password_test.rb

```diff
@@ -31,6 +31,29 @@ class SecurePasswordTest < ActiveModel::TestCase
     assert_not_respond_to @visitor, :valid?
   end
 
+  test "support presence validation options" do
+    class VisitorOrUser < Struct.new(:is_visitor, :password_digest)
+      include ActiveModel::SecurePassword
+      has_secure_password validations: false
+      validates_secure_password unless: :is_visitor
```

Comment:
Hey Jonathan, so this reads weird to me:

If I am reading this correctly, if we want to trigger the validations only on certain conditions we now have to:
1. set the `validations` parameter of `has_secure_password` to false so that it's not triggered automatically
2. specifically call `validates_secure_password` with our condition to trigger it or not.


since `has_secure_password` already has a `validations` parameter, wouldn't it be easier to use that directly ? 

I think I wanted to avoid the whole if/else as well in my original PR by using a Proc (so that all the logic can be done on a method and be called there).

What do you think ?  Anything I am missing here ? 🤔 ( I am also trying to remember what I wanted to do originally, will need to look back at my side project to remember it)

Reply:
I came back to this with a fresh head. 
I like the separation of the validation into a different method. I also think this would fix my issue (or do what I want). 

Still would have prefer not having to do this in multiple steps from a model though - but 🤷 

Not sure who can approve